crates/project/src/terminals.rs

```diff
@@ -31,8 +31,21 @@ impl Project {
             "creating terminals as a guest is not supported yet"
         );
 
+        let cwd = working_directory.as_deref().or_else(|| {
```

Comment:
Sorry, I have confused this part with the https://github.com/zed-industries/zed/pull/11526#issuecomment-2100016642 one, but seems that I've misunderstood that a bit.

This `cwd` is possible to set in the task template, and the user who set the task, might expect the task to run in the specific override, not in the working directory we've derived out of the current project data for regular terminal tab.
Does it make sense? What were the issues causing this change?

Reply:
> This `cwd` is possible to set in the task template, and the user who set the task, might expect the task to run in the specific override, not in the working directory we've derived out of the current project data for regular terminal tab.

This `cwd` is only for locating the worktree that the task is to run in. If there's no working directory from the project, is it not OK to try to find one from the cwd set in the task? If there's none there either, then it'll just end up `None` and using the global terminal settings.

Reply:
I see, good point on the 

> for locating the worktree

I've overlooked it — in that case, should we even consider task's `cwd` for such resolutions then?
I do not have a good answer and maybe it's fine, but indeed seems more logical to prefer the overridden config first with this set-up?
I would entirely ignore task's cwd when looking for the settings, but not certain.

One thing I'm certain about with the current model: we want to find the `SettingsLocation` for both paths, just in case there's an odd situation when we might find no settings for the first path, bit find something for the 2nd one.
And if, on practice, this does not matter much with the current code, it starts to matter if we inverse the precedence.

Whatever approach we take, it might be good to emphasize it by extracting into a method with a good name, or at least leave the comment that task's cwd in this context is used a bit differently (as when spawning the tasks, this `cwd` always overrides other values)

Reply:
> should we even consider task's `cwd` for such resolutions then?

I think if a task is executing with a CWD that resides within a particular worktree you probably want to use that worktree's shell. In general I'd prefer that assumption over failing to find the right shell even if the task runs with a CWD that certainly has an override.

> indeed seems more logical to prefer the overridden config first with this set-up

> One thing I'm certain about with the current model: we want to find the SettingsLocation for both paths, just in case there's an odd situation when we might find no settings for the first path, bit find something for the 2nd one.

Interesting. So if the terminal's working directory doesn't have an override but the task's set to spawn with a CWD that does have an override, you want to use the override anyway? I can certainly do that.

> it starts to matter if we inverse the precedence

+1

> Whatever approach we take, it might be good to emphasize it by extracting into a method with a good name, or at least leave the comment that task's cwd in this context is used a bit differently (as when spawning the tasks, this `cwd` always overrides other values)

Will make sure to throw this change in at some point.

Reply:
The cwd stuff is now inside of a block expression which should make it abundantly clear where they stop being used. Hopefully that will prevent any potential confusion :)